src/main/kotlin/com/example/realworldkotlinspringbootjdbc/domain/ArticleRepository.kt

```diff
@@ -40,4 +40,19 @@ interface ArticleRepository {
     sealed interface TagsError : MyError {
         data class Unexpected(override val cause: Throwable) : TagsError, MyError.MyErrorWithThrowable
     }
+
+    /**
+     * 記事をお気に入りに追加
+     *
+     * @param slug お気に入りにしたい記事の slug
+     * @param currentUserId ログインユーザー
+     * @return 準正常系:エラー or 正常系:お気に入り
+     */
+    fun favorite(slug: Slug, currentUserId: UserId): Either<FavoriteError, CreatedArticle> = TODO()
+    sealed interface FavoriteError : MyError {
+        data class ArticleNotFoundBySlug(val slug: Slug) : FavoriteError, MyError.Basic
```

Comment:
[nits]

```suggestion
        data class NotFoundCreatedArticleBySlug(val slug: Slug) : FavoriteError, MyError.Basic
```

`NotFound〇〇` の方を提案です

Reply:
こちら修正しました

https://github.com/sunakan/realworld-kotlin-springboot-jdbc/pull/110/commits/cbbeed84fe9ce8287f6a14eafd96b4e8faeff625